src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs

```diff
@@ -197,27 +197,28 @@ private void AddDeleteOperation(OpenApiPathItem item, NavigationPropertyRestrict
             Debug.Assert(!LastSegmentIsRefSegment);
 
             DeleteRestrictionsType navPropDeleteRestriction = restriction?.DeleteRestrictions ??
-               Context.Model.GetRecord<DeleteRestrictionsType>(NavigationProperty);
-
+               Context.Model.GetRecord<DeleteRestrictionsType>(NavigationProperty);
+
             if (NavigationProperty.ContainsTarget)
             {
-                DeleteRestrictionsType entityDeleteRestriction = Context.Model.GetRecord<DeleteRestrictionsType>(_navPropEntityType);
-                bool isDeletableDefault = navPropDeleteRestriction == null && entityDeleteRestriction == null;
-
-                if (isDeletableDefault ||
-                ((entityDeleteRestriction?.IsDeletable ?? true) &&
-                (navPropDeleteRestriction?.IsDeletable ?? true)))
-                {
-                    if (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many || LastSegmentIsKeySegment)
-                    {
-                        AddOperation(item, OperationType.Delete);
-                    }
-                }
+                DeleteRestrictionsType entityDeleteRestriction = Context.Model.GetRecord<DeleteRestrictionsType>(_navPropEntityType);
+                bool isDeletableDefault = navPropDeleteRestriction == null && entityDeleteRestriction == null;
+
+                if ((isDeletableDefault ||
+                ((entityDeleteRestriction?.IsDeletable ?? true) &&
+                (navPropDeleteRestriction?.IsDeletable ?? true))) &&
+                (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many ||
+                LastSegmentIsKeySegment))
+                {
+                    AddOperation(item, OperationType.Delete);
+                }
             }
             else
             {
-                if ((navPropDeleteRestriction?.IsDeletable ?? false) &&
-                    (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many ||
+                // Add delete operation for non-contained nav. props only if explicitly set to true via annotation
+                // Note: Use Deletable and not IsDeletable
+                if ((navPropDeleteRestriction?.Deletable ?? false) && 

```

Comment:
collapse this condition into the parent else

Reply:
Resolved